WORKSPACE can point to any place

I noticed that some code from SBSA ACS is now in BSA ACS. And while BSA ACS builds fine, SBSA ACS stopped building (gcc command split to show all arguments in easier to read way):

```
Building ... /home/marcin/devel/linaro/sbsa-qemu/code/edk2/ShellPkg/Application/sbsa-acs/uefi_app/SbsaAvs.inf [AARCH64]
make[1]: Wejście do katalogu '/home/marcin/devel/linaro/sbsa-qemu/code/Build/Shell/DEBUG_GCC49/AARCH64/ShellPkg/Application/sbsa-acs/uefi_app/SbsaAvs'
"aarch64-linux-gnu-gcc" 
-MMD 
-MF /home/marcin/devel/linaro/sbsa-qemu/code/Build/Shell/DEBUG_GCC49/AARCH64/ShellPkg/Application/sbsa-acs/uefi_app/SbsaAvs/OUTPUT/SbsaAvsMain.obj.deps  
-g 
-Os 
-fshort-wchar 
-fno-builtin 
-fno-strict-aliasing 
-Wall 
-Werror 
-Wno-array-bounds 
-include AutoGen.h 
-fno-common 
-ffunction-sections 
-fdata-sections 
-DSTRING_ARRAY_NAME=SbsaStrings 
-g 
-Os 
-fshort-wchar 
-fno-builtin 
-fno-strict-aliasing 
-Wall 
-Werror 
-Wno-array-bounds 
-include AutoGen.h 
-fno-common 
-mlittle-endian 
-fno-short-enums 
-fverbose-asm 
-funsigned-char 
-ffunction-sections 
-fdata-sections 
-Wno-address 
-fno-asynchronous-unwind-tables 
-fno-unwind-tables 
-fno-pic 
-fno-pie 
-ffixed-x18 
-mcmodel=small 
-O0 
-O0 
-I/home/marcin/devel/linaro/sbsa-qemu/code/ShellPkg/Application/bsa-acs/val/ 
-I/home/marcin/devel/linaro/sbsa-qemu/code/ShellPkg/Application/bsa-acs/val/sbsa 
-I/home/marcin/devel/linaro/sbsa-qemu/code/ShellPkg/Application/bsa-acs/val/common 
-I/home/marcin/devel/linaro/sbsa-qemu/code/ShellPkg/Application/bsa-acs/ 
-D DISABLE_NEW_DEPRECATED_INTERFACES 
-c 
-o /home/marcin/devel/linaro/sbsa-qemu/code/Build/Shell/DEBUG_GCC49/AARCH64/ShellPkg/Application/sbsa-acs/uefi_app/SbsaAvs/OUTPUT/./SbsaAvsMain.obj 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/ShellPkg/Application/sbsa-acs/test_pool/ete/operating_system 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/ShellPkg/Application/sbsa-acs/test_pool/ras/operating_system 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/ShellPkg/Application/sbsa-acs/test_pool/pmu/operating_system 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/ShellPkg/Application/sbsa-acs/test_pool/mpam/operating_system 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/ShellPkg/Application/sbsa-acs/test_pool/exerciser/operating_system 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/ShellPkg/Application/sbsa-acs/test_pool/memory_map/operating_system 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/ShellPkg/Application/sbsa-acs/test_pool/smmu/operating_system 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/ShellPkg/Application/sbsa-acs/test_pool/pcie/operating_system 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/ShellPkg/Application/sbsa-acs/test_pool/watchdog/operating_system 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/ShellPkg/Application/sbsa-acs/test_pool/gic/operating_system 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/ShellPkg/Application/sbsa-acs/test_pool/pe/operating_system 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/ShellPkg/Application/sbsa-acs 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/ShellPkg/Application/sbsa-acs/uefi_app 
-I/home/marcin/devel/linaro/sbsa-qemu/code/Build/Shell/DEBUG_GCC49/AARCH64/ShellPkg/Application/sbsa-acs/uefi_app/SbsaAvs/DEBUG 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2-libc/StdLib 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2-libc/StdLib/Include 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2-libc/StdLib/Include/Aarch64 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/EmbeddedPkg 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/EmbeddedPkg/Include 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/EmbeddedPkg/Test/Mock/Include 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/MdePkg 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/MdePkg/Include 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/MdePkg/Test/UnitTest/Include 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/MdePkg/Test/Mock/Include 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/MdePkg/Library/MipiSysTLib/mipisyst/library/include 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/MdePkg/Include/AArch64 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/MdeModulePkg 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/MdeModulePkg/Test/Mock/Include 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/MdeModulePkg/Include 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/ShellPkg 
-I/home/marcin/devel/linaro/sbsa-qemu/code/edk2/ShellPkg/Include 
/home/marcin/devel/linaro/sbsa-qemu/code/edk2/ShellPkg/Application/sbsa-acs/uefi_app/SbsaAvsMain.c
/home/marcin/devel/linaro/sbsa-qemu/code/edk2/ShellPkg/Application/sbsa-acs/uefi_app/SbsaAvsMain.c:26:10: fatal error: val/sbsa/include/sbsa_val_interface.h: Nie ma takiego pliku ani katalogu
   26 | #include "val/sbsa/include/sbsa_val_interface.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [GNUmakefile:1146: /home/marcin/devel/linaro/sbsa-qemu/code/Build/Shell/DEBUG_GCC49/AARCH64/ShellPkg/Application/sbsa-acs/uefi_app/SbsaAvs/OUTPUT/SbsaAvsMain.obj] Błąd 1
make[1]: Opuszczenie katalogu '/home/marcin/devel/linaro/sbsa-qemu/code/Build/Shell/DEBUG_GCC49/AARCH64/ShellPkg/Application/sbsa-acs/uefi_app/SbsaAvs'
```

I looked at INF files and noticed that you assume that WORKSPACE variable points to edk2 checkout (again split to multiple lines):

```
  GCC:*_*_*_CC_FLAGS   = -O0 -I${WORKSPACE}/ShellPkg/Application/bsa-acs/val/
 -I${WORKSPACE}/ShellPkg/Application/bsa-acs/val/sbsa
 -I${WORKSPACE}/ShellPkg/Application/bsa-acs/val/common
 -I${WORKSPACE}/ShellPkg/Application/bsa-acs/
```

Can it get done in a way which does not requires WORKSPACE to be set to edk2/ checkout?

--- a/tools/scripts/acsbuild.sh
+++ b/tools/scripts/acsbuild.sh
@@ -22,6 +22,19 @@ then
     return 0
 fi
 
+# Get the path of the current shell script. Based on the script path navigate to sbsa-acs path
+script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+
+sbsa_path="$(dirname "$(dirname "$script_dir")")"
+echo "sbsa-acs path is: $(realpath "$sbsa_path")"
+
+# Use the default bsa-acs directory
+bsa_path="$sbsa_path/../bsa-acs"
+
+# Export BSA_PATH to bsa-acs path. This will be used in .inf files with -I compiler option.
+export BSA_PATH=$(realpath "$bsa_path")
+echo "bsa-acs path set to: $(realpath "$bsa_path")"
+
 NISTStatus=1;
 
 function build_with_NIST()

--- a/uefi_app/SbsaAvs.inf
+++ b/uefi_app/SbsaAvs.inf
@@ -227,4 +227,4 @@
 
 [BuildOptions]
   GCC:*_*_*_ASM_FLAGS  =  -march=armv8.1-a
-  GCC:*_*_*_CC_FLAGS   = -O0 -I${WORKSPACE}/ShellPkg/Application/bsa-acs/val/ -I${WORKSPACE}/ShellPkg/Application/bsa-acs/val/sbsa -I${WORKSPACE}/ShellPkg/Application/bsa-acs/val/common -I${WORKSPACE}/ShellPkg/Application/bsa-acs/
+  GCC:*_*_*_CC_FLAGS   = -O0 -I${BSA_PATH}/ -I${BSA_PATH}/val/ -I${BSA_PATH}/val/sbsa -I${BSA_PATH}/val/common

--- a/uefi_app/SbsaAvsNist.inf
+++ b/uefi_app/SbsaAvsNist.inf
@@ -231,4 +231,4 @@
 
 [BuildOptions]
   GCC:*_*_*_ASM_FLAGS  =  -march=armv8.1-a
-  GCC:*_*_*_CC_FLAGS   = -O0 -I${WORKSPACE}/ShellPkg/Application/bsa-acs/val/ -I${WORKSPACE}/ShellPkg/Application/bsa-acs/val/sbsa -I${WORKSPACE}/ShellPkg/Application/bsa-acs/val/common -I${WORKSPACE}/ShellPkg/Application/bsa-acs/
+  GCC:*_*_*_CC_FLAGS   = -O0 -I${BSA_PATH}/ -I${BSA_PATH}/val/ -I${BSA_PATH}/val/sbsa -I${BSA_PATH}/val/common

